src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -3838,11 +3838,29 @@ private static bool CanReuseParameter(CSharp.Syntax.ParameterSyntax parameter)
                 type = null;
                 name = this.EatToken(SyntaxKind.ArgListKeyword);
             }
-            var exclamation = this.CurrentToken.Kind == SyntaxKind.ExclamationToken ? this.EatToken(SyntaxKind.ExclamationToken) : null;
+            SyntaxToken exclamation = null;
+            SyntaxToken equals = null;
```

Comment:
consider not giving these initial values, forcing it to be assigned by all paths of the if below.

Reply:
Seems wordier but pushed